core/gethwrappers/abigen.go

```diff
@@ -264,6 +264,41 @@ func addContractStructFields(contractName string, fileNode *ast.File) *ast.File
 		return false
 	}, nil).(*ast.File)
 
+	// Add the address and abi fields to the returned struct in the 'Deploy<contractName>' method.
+	fileNode = astutil.Apply(fileNode, func(cursor *astutil.Cursor) bool {
+		x, is := cursor.Node().(*ast.FuncDecl)
+		if !is {
+			return true
+		} else if x.Name.Name != "Deploy"+contractName {
+			return false
+		}
+
+		for _, stmt := range x.Body.List {
+			returnStmt, is := stmt.(*ast.ReturnStmt)
+			if !is {
+				continue
+			}
+			if len(returnStmt.Results) < 3 {
+				continue
+			}
+			lit, is := returnStmt.Results[2].(*ast.UnaryExpr).X.(*ast.CompositeLit)
```

Comment:
Could the first cast to `*ast.UnaryExpr` fail?

Reply:
I added a check to this type assertion.
Also refactored the code a bit.